boot/boot_serial/src/boot_serial.c

```diff
@@ -339,13 +347,57 @@ bs_upload(char *buf, int len)
             img_blen -= rem_bytes;
         }
     }
+
+#ifdef CONFIG_BOOT_ERASE_PROGRESSIVELY
+    static off_t off_last = -1;
+    struct flash_sector sector;
```

Comment:
To maintain the current style, please move those declarations to the functions' head...